crates/pst/contract/definition/src/state.rs

```diff
@@ -17,6 +18,15 @@ pub struct PstState {
     pub balances: HashMap<String, u64>,
 }
 
+impl Evolve for PstState {
```

Comment:
How will the contract authors know that they need to implement this trait?

I thought about sth like:

```
pub struct PstState {
    #[serde(flatten)]
    pub evolve: EvolveState,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    pub owner: String,
    pub balances: HashMap<String, u64>,
}
```

- the evolve would be a struct that could be composed with contract-specific state...

or maybe a macro?
```
#[warp-contract(evolve)]
pub struct PstState {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    pub owner: String,
    pub balances: HashMap<String, u64>,
}
```

- that automagically adds required fields to the State struct and also implements the `Evolve` trait (which I believe will always look the same, ie. will have the same mapping:
```
can_evolve: self.can_evolve.unwrap_or(true),
source_transaction_id: self.evolve.clone(),
```
)

Reply:
compilation will fail saying that required trait is not implemented.